src/utils.jl

```diff
@@ -150,6 +150,7 @@ function fsmi_generator(world::UInt, source::LineNumberNode, passtype, fsmitype:
     mi, ci, valid_worlds = try
       code_typed_by_type(tt; world, optimize=false)
     catch err # inference failed, return generic type
+      Core.println("inference of a @resumable function failed -- a slower fallback will be used and everything will still work, however please consider reporting this to the developers of ResumableFunctions.jl so that we can debug and increase performance")
```

Comment:
Should this be a warning instead?

Reply:
This happens during inference passes so there seem to be some limitations to what types of printing can be done. I was following the convention that was already established from the next line, but I am not completely sure about the exact limitations. I will look around for advice and make it neater if possible.

Reply:
From Slack:
![image](https://github.com/JuliaDynamics/ResumableFunctions.jl/assets/705248/819eb66a-1341-4e38-b119-075cfc6186bc)

The reason is relatively simple: default printing and logging can lead to task switches which causes problems in the generators of generated functions.

GPUCompiler.jl though has a neat workaround that retains all the neat logging capabilities. I will port it over so that these can be done in a cleaner fashion:  https://github.com/JuliaGPU/GPUCompiler.jl/blob/fd8f36d07eb72a1a4ba601e5da5f95bf287903e1/src/utils.jl#L55